ios/browser/api/favicon/favicon_attributes.mm

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#import "brave/ios/browser/api/favicon/favicon_attributes.h"
+#include "ios/chrome/common/ui/favicon/favicon_attributes.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+// MARK: - Implementation
+
+@interface BraveFavIconAttributes ()
+@property(nonatomic, strong, readonly) FaviconAttributes* attributes;
+@end
+
+@implementation BraveFavIconAttributes
+- (instancetype)initWithAttributes:(FaviconAttributes*)attributes {
+  if ((self = [super init])) {
+    _attributes = attributes;
```

Comment:
Doesn't seem like there's any reason to hold onto the `FaviconAttributes` pointer, and instead we should just set all the Obj-C properties on init